sw/device/tests/entropy_src_edn_reqs_test.c

```diff
@@ -45,6 +45,17 @@ enum {
   kFpgaLoop = 5,
 };
 
+#ifndef SIVAL_ROM_EXT
```

Comment:
Hi @pamaury, a couple of comments:

1. The following was added to perform a more dynamic test configuration based on the `keymgr` state:

https://github.com/lowRISC/opentitan/blob/dcaf19e3feee5ef4947c7ce8fd457057a3ae2b7c/sw/device/lib/testing/keymgr_testutils.c#L111-L131

2. The following was added to give test cases the ability to detect in which boot stage they are running:

https://github.com/lowRISC/opentitan/blob/dcaf19e3feee5ef4947c7ce8fd457057a3ae2b7c/sw/device/lib/arch/boot_stage.h#L28-L47

Reply:
The approach followed in this PR can be implemented using the `boot_stage.h` approach.

Reply:
Regarding the `keymgr`, I am not sure the approach you are suggesting applies: the test perform a keymgr transition. The issue is not the startup of the keymgr but that the expected stage is different between the ROM_EXT and Owner stages.

Reply:
Hi @pamaury, 

I added some comments on this here: https://github.com/lowRISC/opentitan/pull/22708#discussion_r1578290691

Short answer is I believe so.